sos/collector/sosnode.py

```diff
@@ -781,6 +781,15 @@ def execute_sos_command(self):
                 for line in res['stdout'].splitlines():
                     if fnmatch.fnmatch(line, '*sosreport-*tar*'):
                         path = line.strip()
+                    if line.startswith(" sha256\t"):
+                        self.manifest.add_field('checksum', line.split("\t")[1])
+                        self.manifest.add_field('checksum_type', 'sha256')
+                    if line.startswith(" md5\t"):
+                        self.manifest.add_field('checksum', line.split("\t")[1])
+                        self.manifest.add_field('checksum_type', 'md5')
+                    if line.startswith("The checksum is: "):
+                        self.manifest.add_field('checksum', line.split()[3])
+                        self.manifest.add_field('checksum_type', 'unspecified_guess_by_size')
```

Comment:
This should be automatically set, as the hash sizes are well-known. E.G. something like:

```python
if line.startswith((" sha256\t", " md5\t")):
    checksum = line.split("\t")[1]
if line.startswith("The checksum is: "):
    checksum = line.split()[3]
self.manifest.add_field('checksum', checksum)
self.manifest.add_field('checksum_type', 'md5' if len(checksum) == 32 else 'sha256')
```